File: template/node/pkg/openapi.js
const bot = require('qq-guild-bot');
const request = require('request');
const fs = require('fs');

exports.OpenAPI = function () {
    return bot.createOpenAPI({
        appID: process.env.APP_ID,
        token: process.env.APP_TOKEN,
        sandbox: process.env.SANDBOX_BOT === "true",
    });
}

exports.UploadPic = function (name, file) {
    return new Promise(resolve => {
        const options = {
            'method': 'POST',
            'url': process.env.HOME_URL + '/api/v1/open/guild/pic',
            'headers': {
                'X-AppId': process.env.APP_ID,
                'X-Sandbox': process.env.SANDBOX_BOT === "true" ? '1' : '',
                'X-AppToken': process.env.APP_TOKEN,
            },
            formData: {
                'pic': {
                    'value': file,
                    'options': {
                        'filename': name,
                        'contentType': "image/png"
                    }
                }
            }
        };
        request(options, function (error, response) {
            if (error) throw new Error(error);
            resolve(JSON.parse(response.body));
        });
    });
}

File: nonebot/pkg/openapi.py
import json
import requests
import os
import qqbot

token = qqbot.Token(os.environ.get("APP_ID"), os.environ.get("APP_TOKEN"))
sandbox = os.environ.get("SANDBOX_BOT") == "true"


def OpenAPI():
    return token, sandbox


def PicUpload(name, file):
    url = os.environ.get("HOME_URL")+"/api/v1/open/guild/pic"

    payload = {}
    files = [
        ('pic', (name, file, 'image/png'))
    ]
    sandbox = 0
    if os.environ.get("SANDBOX_BOT") == "true":
        sandbox = 1
    headers = {
        'X-AppId': os.environ.get("APP_ID"),
        'X-Sandbox': str(sandbox),
        'X-AppToken': os.environ.get("APP_TOKEN")
    }

    response = requests.request(
        "POST", url, headers=headers, data=payload, files=files)

    return json.loads(response.text)
